pkg/mutation/assign_mutator.go

```diff
@@ -37,8 +39,41 @@ func (m *AssignMutator) Matches(obj runtime.Object, ns *corev1.Namespace) bool {
 	return matches
 }
 
+func (m *AssignMutator) gatherPathTests() ([]patht.Test, error) {
+	pts := m.assign.Spec.Parameters.PathTests
+	var pathTests []patht.Test
+	for _, pt := range pts {
+		p, err := parser.Parse(pt.SubPath)
+		if err != nil {
+			return nil, err
+		}
+		pathTests = append(pathTests, patht.Test{SubPath: p, Condition: pt.Condition})
+	}
+	return pathTests, nil
+}
+
+func (m *AssignMutator) getTester() (*patht.Tester, error) {
+	if m.tester != nil {
+		return m.tester, nil
```

Comment:
What's the concurrency story for this mutator? Do we need to guard here?

Reply:
I don't think this should be accessed in a concurrent manner, but no reason not to throw a lock around this.

Reply:
Added an RWMutex

Reply:
I can definitely add the second `nil` check.

WRT sync.Once, I think we probably want to run the code every time if the value is nil, so that we can continue to return an appropriate error if `nil` is the result of an error.

The other option would be to cache the error as well, but that has the potential to make otherwise transient bugs sticky.

Reply:
It sounds like we have a path forward here but removing lazy initialization.

Reply:
Yep! Committed a change that switches to eager initialization, should be good now.